docs/DeveloperGuide.md

```diff
@@ -296,82 +296,131 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
+2.  CLIpboard updates the course with the new name provided by the user
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty
+* 1a. The list is empty
+
+    * 1a1. CLipboard shows an error message.
+
+      Use case ends.
 
-  Use case ends.
+* 1b. The course index does not exist in the list
 
-* 3a. The student does not exist in the list
+    * 1b1. CLipboard shows an error message.
 
-    * 3a1. CLipboard shows an error message.
+      Use case ends.
 
-      Use case resumes at step 2.
+<h3>Use case 3: Delete en existing course</h3>
 
-**Use case: Delete a student**
+**Preconditions: User is on the Course Page**
 
 **MSS**
 
-1.  User requests to list students
-2.  CLIpboard shows a list of students
-3.  User requests to delete a specific student in the list
-4.  CLIpboard deletes the student
+1.  User requests to delete a course in the list
+2.  CLIpboard updates the course list without the course
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty
+* Similar to UC2 extension
 
-  Use case ends.
+<h3>Use case 4: Add a new student</h3>
 
-* 3a. The given index is invalid
+**Preconditions: User is on the Student Page**
+
+**MSS**
+
+1.  User requests to add a new student into the list
+2.  CLIpboard adds the particulars of the student into the list
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The list is empty
+  * 1a1. CLIpboard shows an error message.
+
+    Use case ends.
 
-    * 3a1. CLIpboard shows an error message.
+* 1b. A student with the same student ID already exists
 
-      Use case resumes at step 2.
+    * 1b1.  CLIpboard shows an error message.
 
-**Use case: Leave a note under a student's profile**
+      Use case ends.
+
+<h3>Use case 5: Edit an existing student's information</h3>
+
+**Preconditions: User is on the Student Page**
 
 **MSS**
 
-1.  User requests to list students
-2.  CLIpboard shows a list of students
-3.  User requests to add a note under a specific student in the list
-4.  CLIpboard updates the particulars of the student to include the note
+1.  User requests to edit a specific student in the list
+2.  CLIpboard updates the particulars of the student
 
     Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty
+* Similar to UC2 extension, for student index
+
+<h3>Use case 6: Delete a student</h3>
+
+**Preconditions: User is on the Student Page**
+
+**MSS**
 
-  Use case ends.
+1.  User requests to delete a specific student in the list
+2.  CLIpboard deletes the student
 
-* 3a. The student does not exist in the list
+    Use case ends.
 
-    * 3a1. CLipboard shows an error message.
+**Extensions**
 
-      Use case resumes at step 2.
+* Similar to UC2 extension, for student index
+
+**Use case 7: Update a remark under a student's profile**
+
+**Preconditions: User is on the Student Page**
+
+**MSS**
+
+3.  User requests to update a remark under a specific student in the list
+4.  CLIpboard updates the particulars of the student to the new remark
```

Comment:
Is it intended to have numbering or 3 and 4 here?

Reply:
Thanks for catching this! Updated the numbering to 1 and 2.